src/main/java/healthcare/aura/websocketserver/AuraWebSocketServer.java

```
@@ -69,8 +75,28 @@ public void onMessage(WebSocket conn, String message) {
         LOGGER.info("###### Thread number : {}", Thread.currentThread().getId());
 
         Gson gson = new Gson();
-        SensorData sensorData = gson.fromJson(message, SensorData.class);
-        String fileName = sensorData.generateFileName();
+        String fileName = "";
+
+        try{
+          SensorData sensorData = gson.fromJson(message, SensorData.class);
+          fileName = sensorData.generateFileName();
+        } catch (Exception e){
+          LOGGER.info("Not a sensor data file");
+        }
+
+        try{
+          Type type = new TypeToken<ArrayList<SensitiveEventData> >(){}.getType();
+          ArrayList<SensitiveEventData> seizureDataList = gson.fromJson(message, type);
+          fileName = seizureDataList.get(0).generateFileName();
+          LOGGER.info("FileName " + fileName);
+        } catch (Exception e){
+          LOGGER.info("Not a seizure data file");
+        }
+
+        if(fileName.isEmpty()){
+          LOGGER.error("Invalid input data file");
+          return;
```

Throw a custom exception rather than return ?